apps/ex_nvr/lib/ex_nvr/remote_storages/store.ex

```diff
@@ -0,0 +1,22 @@
+defmodule ExNVR.RemoteStorages.Store do
+  @moduledoc false
+
+  alias ExNVR.Model.{Device, Recording}
+  alias ExNVR.RemoteStorages.Store.{HTTP, S3}
+
+  @callback save_recording(Device.t(), Recording.t(), opts :: Keyword.t()) ::
+              :ok | {:error, any()}
+
+  @callback save_snapshot(
+              Device.t(),
+              snapshot :: binary(),
+              timestamp :: DateTime.t(),
+              opts :: Keyword.t()
+            ) :: :ok | {:error, any()}
+
+  def save_snapshot(device, snapshot, timestamp, opts),
```

Comment:
I prefer you pass a remote storage as first argument to this function instead of passing the type in the opts